File: 459_RepeatedSubstringPattern.cpp
//28.53%
class Solution {
public:
    bool repeatedSubstringPattern(string str) {
        int size(str.size());
		for(int i = 1; i <= size / 2; i++){
			if(size % i == 0){
				string temp(str.begin(), str.begin() + i);
				string res;
				for(int j = 0; j < size / i; j++){
					res += temp;
				}
				if(res == str)
					return true;
			}
		}
		return false;
    }
};

File: 459_RepeatedSubstringPattern.py
class Solution(object):
    def repeatedSubstringPattern(self, str):
        """
        :type str: str
        :rtype: bool
        """
        size = len(str)
        for i in range(1, size / 2 + 1):
            if size % i == 0:
                item = str[:i] * (size / i)
                if item == str:
                    return True
        return False
		
def repeatedSubstringPattern(self, str):

        """
        :type str: str
        :rtype: bool
        """
        if not str:
            return False
            
        ss = (str + str)[1:-1]
        return ss.find(str) != -1